Bicep compilation failed for Bicep Linter Configuration info message

**Description of the issue**

Bicep compilation failed for Bicep Linter Configuration info message microsoft/PSRule-pipelines#266

**Info** and **Warning** messages should not be disabled in errors.

**Error output**

```text
##[error]Bicep (0.4.1008) compilation of '/home/vsts/work/1/s/adfrowpe.bicep' failed with: /home/vsts/work/1/s/adfrowpe.bicep(1,1) : Info Bicep Linter Configuration: Custom bicepconfig.json file found (/home/vsts/work/1/s/bicepconfig.json).

```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **v1.11.0-B2111014**

--- a/docs/CHANGELOG-v1.md
+++ b/docs/CHANGELOG-v1.md
@@ -7,6 +7,11 @@ See [troubleshooting guide] for a workaround to this issue.
 
 ## Unreleased
 
+What's changed since pre-release v1.11.0-B2112073:
+
+- Bug fixes:
+  - Fixed output of Bicep informational and warning messages in error stream. [#1157](https://github.com/Azure/PSRule.Rules.Azure/issues/1157)
+
 ## v1.11.0-B2112073 (pre-release)
 
 What's changed since pre-release v1.11.0-B2112024:

--- a/src/PSRule.Rules.Azure/Data/Bicep/BicepHelper.cs
+++ b/src/PSRule.Rules.Azure/Data/Bicep/BicepHelper.cs
@@ -173,8 +173,21 @@ namespace PSRule.Rules.Azure.Data.Bicep
                 }
                 else
                 {
-                    _Error.AppendLine(e.Data);
+                    var errors = GetErrorLine(e.Data);
+                    for (var i = 0; i < errors.Length; i++)
+                        _Error.AppendLine(errors[i]);
                 }
+            }
+
+            private static string[] GetErrorLine(string input)
+            {
+                var lines = input.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+                var result = new List<string>();
+                for (var i = 0; i < lines.Length; i++)
+                    if (!lines[i].Contains(": Warning ") && !lines[i].Contains(": Info "))
+                        result.Add(lines[i]);
+
+                return result.ToArray();
             }
 
             private void Dispose(bool disposing)

--- a/tests/Bicep/Bicep.Tests.ps1
+++ b/tests/Bicep/Bicep.Tests.ps1
@@ -108,5 +108,44 @@ Describe 'Bicep' -Tag 'Bicep' {
                 Remove-Item 'Env:PSRULE_AZURE_BICEP_USE_AZURE_CLI' -Force;
             }
         }
+
+        It 'Bicep error message filtered' {
+            $invokeParams = @{
+                Baseline = 'Azure.All'
+                Module = 'PSRule.Rules.Azure'
+                WarningAction = 'Ignore'
+                ErrorAction = 'Stop'
+            }
+
+            try {
+                # Install CLI
+                az bicep install
+
+                # Expand source files
+                $option = @{
+                    'Configuration.AZURE_BICEP_FILE_EXPANSION' = $True
+                }
+                $Env:PSRULE_AZURE_BICEP_USE_AZURE_CLI = 'true';
+
+                # Process file with errors
+                $sourceFile = Join-Path -Path $here -ChildPath 'error.bicep';
+                $Null = Invoke-PSRule @invokeParams -InputPath $sourceFile -Format File -Option $option -ErrorVariable errors -ErrorAction SilentlyContinue;
+                $bicepError = @($errors | Where-Object {
+                    $_.Exception -is [PSRule.Rules.Azure.Pipeline.BicepCompileException]
+                })
+                $bicepError.Exception.Message | Should -BeLike "*: Error BCP019:*";
+
+                # Process file without errors
+                $sourceFile = Join-Path -Path $here -ChildPath 'noterror.bicep';
+                $Null = Invoke-PSRule @invokeParams -InputPath $sourceFile -Format File -Option $option -ErrorVariable errors -ErrorAction SilentlyContinue;
+                $bicepError = @($errors | Where-Object {
+                    $_.Exception -is [PSRule.Rules.Azure.Pipeline.BicepCompileException]
+                })
+                $bicepError.Length | Should -Be 0;
+            }
+            finally {
+                Remove-Item 'Env:PSRULE_AZURE_BICEP_USE_AZURE_CLI' -Force;
+            }
+        }
     }
 }

--- a/tests/Bicep/bicepconfig.json
+++ b/tests/Bicep/bicepconfig.json
@@ -1,0 +1,8 @@
+{
+    "analyzers": {
+        "core": {
+            "enabled": true,
+            "verbose": true
+        }
+    }
+}

--- a/tests/Bicep/error.bicep
+++ b/tests/Bicep/error.bicep
@@ -1,0 +1,6 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+param name string
+
+var normalName = concat(name, ''))

--- a/tests/Bicep/noterror.bicep
+++ b/tests/Bicep/noterror.bicep
@@ -1,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+param name string = 'abc'
+param location string = resourceGroup().location
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2019-06-01' = {
+  name: name
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  kind: 'StorageV2'
+  properties: {
+    networkAcls: {
+      bypass: 'AzureServices'
+      virtualNetworkRules: []
+      ipRules: []
+      defaultAction: 'Allow'
+    }
+    supportsHttpsTrafficOnly: true
+    encryption: {
+      services: {
+        file: {
+          enabled: true
+        }
+        blob: {
+          enabled: true
+        }
+      }
+      keySource: 'Microsoft.Storage'
+    }
+    accessTier: 'Hot'
+    minimumTlsVersion: 'TLS1_2'
+  }
+}

